.github/workflows/build-and-test.yml

```
@@ -81,53 +44,26 @@ jobs:
           # We need to fetch all branches and commits so that Nx affected has a base to compare against.
           fetch-depth: 0
 
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
         name: Checkout [Default Branch]
         if: ${{ github.event_name != 'pull_request' }}
         with:
           # We need to fetch all branches and commits so that Nx affected has a base to compare against.
           fetch-depth: 0
 
-      - name: Set up dependencies
-        uses: ./.github/actions/setup
+      - name: Derive appropriate SHAs for base and head for `nx affected` commands
+        uses: nrwl/nx-set-shas@v4
 
       - name: Initialize the Nx Cloud distributed CI run
-        run: yarn nx-cloud start-ci-run
+        run: npx nx-cloud start-ci-run --distribute-on="./nx/workflows/dynamic-changesets.yml" --stop-agents-after="e2e"
```

suggestion: Try using the `e2e-ci` tasks. [Configuration docs for the Nx Cypress plugin](https://nx.dev/nx-api/cypress#nxcypress-configuration).

Specify `e2e-ci` both for `nx cloud start-ci-run --stop-agents-after` and `nx affected -t`.

```suggestion
        run: npx nx-cloud start-ci-run --distribute-on="./nx/workflows/dynamic-changesets.yml" --stop-agents-after="e2e-ci"
```

---

@LayZeeDK tried, didn't work. I don't want to mess around with it any longer :D if you have any other suggestion, i'm going to revert the e2e-ci changes